microsoft/testsuites/mshv/mshv_install.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+import re
+from pathlib import Path
+from typing import Any, Dict
+
+from assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+# from lisa.features import SerialConsole
+from lisa.testsuite import TestResult
+from lisa.tools import Dmesg, Ls, Service, Reboot
+from lisa.util import SkippedException, TcpConnectionException, constants
+from lisa.util.shell import wait_tcp_port_ready
+
+
+@TestSuiteMetadata(
+    area="mshv",
+    category="functional",
+    description="""
+    This test suite is to test VM working well after updating MSHV on VM and rebooting.
+    """,
+)
+class MshvHostInstallSuite(TestSuite):
+
+    _test_hvix_file_path = (
+        Path(os.path.dirname(__file__)) / "test_data" / f"hvix64.exe"
+    )
+
+    _test_hvix_file_path_dst = (
+        Path("/boot/efi/Windows/System32") / f"hvix64.exe"
+    )
+
+    _test_kdstub_file_path = (
+        Path(os.path.dirname(__file__)) / "test_data" / f"kdstub.dll"
+    )
+
+    _test_kdstub_file_path_dst = (
+        Path("/boot/efi/Windows/System32") / f"kdstub.dll"
+    )
+
+    _test_lxhvloader_file_path = (
+        Path(os.path.dirname(__file__)) / "test_data" / f"lxhvloader.dll"
+    )
+
+    _test_lxhvloader_file_path_dst = (
+        Path("/boot/efi") / f"lxhvloader.dll"
+    )
+
+    _init_path_dst = (
+        Path("/home/cloud")
+    )
+
+    @TestCaseMetadata(
+        description="""
+        This test case will
+        1. Update to new MSHV components over old ones in a pre-configured MSHV image
+        2. Reboot VM, check that mshv comes up
+
+        The test expects the MSHV binaries to be installed to be placed under lisa/microsoft/testsuites/mshv/test_data
+        before lisa is executed.
+        """,
+        priority=0,
+        timeout=60,  # 60 seconds
+    )
+    def verify_mshv_install_succeeds(
+        self,
+        log: Logger,
+        node: Node,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        # Copy Hvix64.exe, kdstub.dll, lxhvloader.dll into test machine
+        hvix_path = self._init_path_dst / f"hvix64.exe"
+        node.shell.copy(self._test_hvix_file_path, hvix_path)
```

Comment:
You can change the path from a test visible variable. if the variable is not provided, the test case should be skipped, instead of failed.